registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java

```diff
@@ -433,6 +414,28 @@ public MessageDTO process(MessageDTO object) {
 		return object;
 	}
 
+	private void handleForceMerge(MessageDTO object, boolean isTransactionSuccessful, LogDescription description,
+			String registrationId, UinGenResponseDto uinResponseDto,
+			InternalRegistrationStatusDto registrationStatusDto, JSONObject demographicIdentity,
+			boolean isUinAlreadyPresent) throws ApisResourceAccessException, IOException, Exception,
+			VidCreationException, VidServiceFailedException {
+		sendResponseToUinGenerator(registrationId, uinResponseDto.getResponse().getUin(),
+				UINConstants.UIN_UNASSIGNED);
+		String uin = idRepoService.getUinByRid(registrationId,
```

Comment:
This should be possible only if the given rid is the last known rid for the UIN. I dont see a code taking care of this? So lets assume someone updates their data with a new rid. Later an admin re-runs the original registration packet, we want the system not to process that as it contains older data. or is that a functionality of id-repo? if so what PR fixes that?

Reply:
MOSIP-25402  done fix in id-repo also  for  below problem
We see a data mismatch between the packet data and the id repository data for a few packets. To fix these affected records we need to modify the create API to support the modification of data in the ID repository when these affected records are reprocessed.

Reply:
Yes this is taken care in idrepo to ensure we don't overwrite a old rid data